src/util/instanceMigration.tsx

```diff
@@ -0,0 +1,134 @@
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { useEventQueue } from "context/eventQueue";
+import { useInstanceLoading } from "context/instanceLoading";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { queryKeys } from "./queryKeys";
+import { fetchStoragePools } from "api/storage-pools";
+import ItemName from "components/ItemName";
+import { migrateInstance } from "api/instances";
+import { LxdInstance } from "types/instance";
+import { fetchClusterMembers } from "api/cluster";
+import { ReactNode, useState } from "react";
+import { useSettings } from "context/useSettings";
+import { isClusteredServer } from "./settings";
+import { capitalizeFirstLetter } from "./helpers";
+
+export type MigrationType = "cluster member" | "root storage pool" | "";
+
+type Props = {
+  instance: LxdInstance;
+  close: () => void;
+};
+
+export const useInstanceMigration = (props: Props) => {
+  const { instance, close } = props;
+  const toastNotify = useToastNotification();
+  const instanceLoading = useInstanceLoading();
+  const eventQueue = useEventQueue();
+  const queryClient = useQueryClient();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const [type, setType] = useState<MigrationType>(
+    isClustered ? "" : "root storage pool",
+  );
+  const [target, setTarget] = useState("");
+
+  const { data: pools = [] } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(instance.project),
+    enabled: type === "root storage pool",
+  });
+
+  const { data: members = [] } = useQuery({
+    queryKey: [queryKeys.cluster, queryKeys.members],
+    queryFn: fetchClusterMembers,
+    enabled: type === "cluster member",
+  });
```

Comment:
```suggestion
```
We should load the data, when it is needed, not unconditionally in this hook.

Reply:
100%, I wasn't quite done with this part yet since I was planning on waiting for #916 to get merged. tbh I don't think we need to wait anymore though, I'm quite sure that `StoragePoolSelectTable` can be quite easily implemented in a follow up PR

Reply:
these calls will be moved into the respective table components